README.md

```diff
@@ -87,23 +146,61 @@ contract MockBlocklockReceiver is AbstractBlocklockReceiver {
         // Decrypt stored Ciphertext with the decryption key
         plainTextValue = abi.decode(blocklock.decrypt(encryptedValue, decryptionKey), (uint256));
     }
-}
 ```
+> 💡 **Note:** `blocklock.decrypt` automatically verifies the dcipher threshold decryption key for you because of the power of threshold signatures scheme!
+
+#### 5. Deploy the `BlocklockHandler` contract
+Please check the supprted network section to ensure your desired network is supported before deployment. You also need to use the deployed **BlocklockSender (Proxy)** address to initialized your contract.
+
+When deploying, make sure your desired network is supported. Use the deployed `BlocklockSender (Proxy)` address when initializing your contract.
+
+Example in Foundry Script for Filecoin Calibration: 
+```solidity
+address blocklockSenderProxy = 0xfF66908E1d7d23ff62791505b2eC120128918F44;
+MockBlocklockReceiver mockBlocklockReceiver = new MockBlocklockReceiver(address(blocklockSenderProxy));
+console.log("\nMockBlocklockReceiver deployed at: ", address(mockBlocklockReceiver));
+```
+
+#### 6. Off-chain data encoding and encryption
+Use [blocklock.js](https://github.com/randa-mu/blocklock-js) to encode and encrypt data off-chain before calling the smart contract. It ensures data is encrypted using the dcipher threshold public key and specified block height.
+
+Below is a sample JavaScript/TypeScript snippet:
+```Javascript
+// Encode the uint256 value
+const encoder = new SolidityEncoder();
+const msgBytes = encoder.encodeUint256(msg);
+const encodedMessage = getBytes(msgBytes);
```

Comment:
is `getBytes` a built-in? I don't see it mentioned anywhere before

Reply:
It's imported from `ethers.js`

As in:

`import { getBytes } from "ethers";`

@longfeiWan9  we could add it at the top of the JavaScript code block.